index.d.ts

```diff
@@ -0,0 +1,280 @@
+     *
+     * @param {string} tableId The table ID
+     * @returns {Promise<module:dronedeploy/functions-wrapper.Table>} The table
+     */
+    table(tableId: string): Promise<Table>;
+
+  }
+
+  /**
+   * Response returned from a datastore mutation
+   */
+  interface MutationResponse {
+    ok: boolean;
+    error?: any;
+  }
+
+  /**
+   * Interactions with a table.
+   */
+  interface Table {
+    /**
+     * Inserts a row
+     *
+     * @param {string} externalId The external ID
+     * @param row
+     * @returns {Promise<module:dronedeploy/functions-wrapper.MutationResponse>}
+     */
+    addRow(externalId: string, row: any): Promise<MutationResponse>;
+
+    /**
+     * Updates a row
+     *
+     * @param {string} externalId The external ID
+     * @param row
+     * @returns {Promise<module:dronedeploy/functions-wrapper.MutationResponse>}
+     */
+    editRow(externalId: string, row: any): Promise<MutationResponse>;
+
+    /**
+     * Upserts a row
+     *
+     * @param {string} externalId The external ID
+     * @param row
+     * @returns {Promise<module:dronedeploy/functions-wrapper.MutationResponse>}
+     */
+    upsertRow(externalId: string, row: any): Promise<MutationResponse>;
+
+    /**
+     * Retrieves a row of data by ID
+     *
+     * @param {string} externalId The external ID
+     * @returns {Promise<any>} The row of data
+     */
+    getRowByExternalId(externalId: string): Promise<any>;
+  }
+
+  /**
+   * If you are an owner, then you can access additional admin functionality
+   */
+  interface OwnerDatastore extends Datastore {
+
+    /**
+     * Creates a table
+     *
+     * @param {string} appSlug
+     * @param {string} tableName
+     * @param {string} description
+     * @param columnDefinitions
+     * @returns {Promise<string>}
+     */
+    createTable(appSlug: string, tableName: string, description: string, columnDefinitions: ColumnDefinitionInput[]): Promise<string>
+
+    /**
+     * Ensures that a table exists before trying trying to create.
+     *
+     * @param {string} appSlug
+     * @param {string} tableName
+     * @param {string} description
+     * @param columnDefinitions
+     * @returns {Promise<string>}
+     */
+    ensure(appSlug: string, tableName: string, description: string, columnDefinitions?: ColumnDefinitionInput[]): Promise<string>
+  }
+
+  interface ColumnDefinitionInput {
+    input: ColumnDefinition
+  }
+
+  /**
+   * The different types of column definitions.
+   */
+  type ColumnDefinition =
+    DateColumnDefinition | DateTimeColumnDefinition | EmailColummDefinition | NumberColumnDefinition | TextColumnDefinition;
+
+  /**
+   * The string literals for column types.
+   */
+  type ColumnType = "DATE" | "DATETIME" | "EMAIL" | "NUMBER" | "TEXT";
+
+  /**
+   * Common properties on all columns
+   */
+  interface GenericColumnDefinition<T extends ColumnType> {
+    columnType: T;
+    name: string;
+    description: string;
+    nullabe?: string;
```

Comment:
This is going to be a boolean - https://github.com/dronedeploy/drone_web/blob/eae4f2a3cc0e833434944b959b68045d6a83d54a/server/graphql_apis/schema.py#L467